src/TfvcMigrator/Program.cs

```diff
@@ -253,10 +253,16 @@ private static RootPathChange ParseRootPathChange(string token)
 
                     foreach (var (_, parentChangeset, parentBranch) in mappingState.AdditionalParents.Where(t => t.Branch == branch))
                     {
-                        if (commitsByChangeset.TryGetValue(parentChangeset, out var createdChangesets)
-                            && createdChangesets.SingleOrDefault(c => c.Branch == parentBranch).Commit is { } commit)
+                        if (commitsByChangeset.TryGetValue(parentChangeset, out var createdChangesets))
                         {
-                            parents.Add(commit);
+                            if (createdChangesets.SingleOrDefault(c => c.Branch == parentBranch).Commit is { } commit)
+                            {
+                                parents.Add(commit);
+                            }
+                            else
+                            {
+                                throw new InvalidOperationException("Should not be reachable. Should be a single changeset matching parent branch.");
```

Comment:
Don't know what the error message should be here, just thought it is useful if the checks give slightly different errors.

Reply:
I think this is useful. The message for the first check could be: `A commit should have been created for the parent changeset {A} before reaching this point.`

The message for the second check could be: `None of the commits created for the parent changeset {A} has the expected branch {B}.`

Reply:
updated